nutanix/resource_nutanix_ndb_cluster.go

```diff
@@ -0,0 +1,611 @@
+		Delay:   eraDelay,
+	}
+
+	if _, errWaitTask := stateConf.WaitForStateContext(ctx); errWaitTask != nil {
+		return diag.Errorf("error waiting for cluster (%s) to register: %s", resp.Entityid, errWaitTask)
+	}
+
+	clsName := d.Get("name")
+	// api to fetch clusters based on name
+	getResp, er := conn.Service.GetCluster(ctx, "", clsName.(string))
+	if er != nil {
+		return diag.FromErr(er)
+	}
+	d.SetId(*getResp.ID)
+	log.Printf("NDB cluster with %s id is registered successfully", d.Id())
+	return resourceNutanixNDBClusterRead(ctx, d, meta)
+}
+
+func resourceNutanixNDBClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	resp, err := conn.Service.GetCluster(ctx, d.Id(), "")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("id", resp.ID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("name", resp.Name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("unique_name", resp.Uniquename); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("ip_addresses", resp.Ipaddresses); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("fqdns", resp.Fqdns); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("nx_cluster_uuid", resp.Nxclusteruuid); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("description", resp.Description); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cloud_type", resp.Cloudtype); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("date_created", resp.Datecreated); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("date_modified", resp.Datemodified); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("owner_id", resp.Ownerid); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("status", resp.Status); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("version", resp.Version); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("hypervisor_type", resp.Hypervisortype); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("hypervisor_version", resp.Hypervisorversion); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("properties", flattenClusterProperties(resp.Properties)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("reference_count", resp.Referencecount); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("username", resp.Username); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cloud_info", resp.Cloudinfo); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("resource_config", flattenResourceConfig(resp.Resourceconfig)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("management_server_info", resp.Managementserverinfo); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("entity_counts", resp.Entitycounts); err != nil {
```

Comment:
can we follow naming like "EntityCounts" for all above as well

Reply:
For better understanding of struct names

Reply:
Done.